### PR TITLE
Support passive event

### PR DESCRIPTION
**Pull request by roderickhsiao**
The follow up PR based on #43 to address #42 

passing an option to the event listener instead of specifically passing `passive: true` so that we can support other event.

The event object will only be passed when browser support the event object.

@hankhsiao @redonkulus @kaesonho 

*Note*: This PR will turn passive event default to true.

### CODE REVIEW

File: src/globalVars.js [hunk 1]
```diff
@@ -5,12 +5,14 @@
 'use strict';
 
 var EventEmitter = require('eventemitter3');
+var supoortPassiveEvent = require('./lib/supportPassiveEvent');
```
Comment (redonkulus) [inline src/globalVars.js]: typo extra `o`

File: src/lib/listen.js [hunk 2]
```diff
@@ -3,25 +3,34 @@
...
  * @param {String} eventType - The event type.
  * @param {Function} handler - The event handler.
+ * @param {Object} handler - The options object that specifies characteristics about the event listener.
  * @return {Object} The object to be able to remove the handler.
  */
-function listen(target, eventType, handler) {
+function listen(target, eventType, handler, options) {
     var add = 'addEventListener';
     var remove = 'removeEventListener';
+    var eventOptions = globalVars.supportsPassive ? _assign(defaultEventOption, options) : false;
```
Comment (redonkulus) [inline src/lib/listen.js]: minor, but maybe add `{}` as first argument to _assign to avoid changing `defaultEventOption`.

**Comment by roderickhsiao** [comment]
@redonkulus good question about the testing, probably in functional test, but not sure if it is reliable though

**Comment by gyehuda** [comment]
@redonkulus done.

**Comment by roderickhsiao** [comment]
@redonkulus the below unit test could be added 

https://github.com/yahoo/subscribe-ui-event/pull/44/commits/5203eeebe37634b13de7c6c6b80d7f0ba5958c22 

after JSDOM support passive event
https://github.com/tmpvar/jsdom/blob/master/test/web-platform-tests/to-run.yaml#L132
https://github.com/tmpvar/jsdom/issues/1609

**Comment by roderickhsiao** [comment]
I tried to add test in functional test, but unfortunately I think functional test doesn't run for a long time, and its expecting @hankhsiao access token :P

**Comment by hankhsiao** [comment]
It looks good to me 👍

**Comment by redonkulus** [comment]
Thanks!

**Comment by roderickhsiao** [comment]
🎅 🤶

**Comment by redonkulus** [comment]
released in 1.1.0 🎄

**Pull request merged**